site/docs/red-team/plugins/index.md

```diff
@@ -0,0 +1,272 @@
+
+| LLM Design              | Non-Applicable Tests                 |
+| ----------------------- | ------------------------------------ |
+| **Foundation Model**    | Security and Access Control Tests    |
+| **Single User Role**    | Access Control Tests                 |
+| **Prompt and Response** | Resource Fetching, Injection Attacks |
+
+### RAG Architecture and/or Agent Architecture
+
+For LLM applications with agentic or RAG components, it is recommended to test for application-level vulnerabilities:
+
+```yaml
+plugins:
+  - 'rbac' # Tests if the model properly implements Role-Based Access Control
+  - 'bola' # Checks for Broken Object Level Authorization vulnerabilities
+  - 'bfla' # Tests for Broken Function Level Authorization issues
+  - 'ssrf' # Tests for Server-Side Request Forgery vulnerabilities
+  - 'sql-injection' # Tests for SQL injection vulnerabilities (if connected to a SQL database)
+  - 'pii' # Checks for leakage of Personally Identifiable Information
+  - 'excessive-agency' # Checks if the agent exceeds its intended capabilities
+  - 'hijacking' # Checks for goal hijacking of the agent's objectives
+```
+
+### Plugin Collections
+
+You can also select a preset configuration based on common security frameworks and standards.
+
+| Framework                             | Plugin ID       | Example Specification      |
+| ------------------------------------- | --------------- | -------------------------- |
+| **NIST AI Risk Management Framework** | nist:ai:measure | owasp:llm:01               |
+| **OWASP Top 10 for LLMs**             | owasp:llm       | nist:ai:measure:1.1        |
+| **OWASP Top 10 for APIs**             | owasp:api       | owasp:api:01               |
+| **MITRE ATLAS**                       | mitre:atlas     | mitre:atlas:reconnaissance |
+
+## Plugin Descriptions
+
+Plugins fall into five categories: security and access control, compliance and legal, trust and safety, brand, and custom plugins.
+
+- Security and access control plugins focus on technical security risk, such as SQL injection, SSRF, broken access control, and cross-session leaks. These are closely mapped to the OWASP Top 10 for LLMs, APIs, and web applications.
+- Compliance and legal plugins focus on LLM behavior that may encourage illegal activity, breach contractual commitments, or violate intellectual property rights.
+- Trust and safety plugins attempt to product illicit, graphic, or inappropriate responses from the LLM.
+- Custom plugins can be established for testing your policies or generating specific probes for your use case.
+
+Some plugins point to your own LLM provider to generate adversarial probes. Others must point to Promptfoo's remote generation endpoint.
+
+| Category                        | Strategy                                                  | Description                                                                             | Plugin ID                             |
```

Comment:
This is the third place we're creating a list like this, after configuration.md and llm-vulnerability-types.md.  Should we make a single source of truth/worth creating a single table template that we can just include where necessary?

Reply:
Cool. Where should keep the single source of truth? My preference would be in this page (plugins/index.md). Let me know and I can change accordingly.